### ISSUE TITLE
Boost?

### ISSUE THREAD

**Issue by ibell** [CoolProp/CoolProp#1789]
@jowr - any strong opinions about adding boost to our stack?  It would massively simplify the treatment of paths, amongst other things - that's what i did with the REFPROP testing framework: https://github.com/usnistgov/REFPROP-tests.  With this (https://github.com/Orphis/boost-cmake) it is dead-simple to integrate into a cmake-based build, like in CoolProp.

**Comment by jowr** [comment]
No problem with boost. The only issue is that we have to make a million changes in many files. Whatever we do, I would still advocate for using hunter as package manager. It really speeds up build time on local machines.

**Comment by ibell** [comment]
Yeah, I know, I plan to start small, for instance to migrate all filesystem stuff to use ``boost::filesystem``.  Our homegrown code for that seems to be rather fragile, and with boost it would be trivial to do things that are quite involved in our code.

The downside with hunter as I see it is that we have to use CMake for everything.  In general we do, of course the big exception being Python.

**Comment by ibell** [comment]
Closing as part of an issue-tracker cleanup — this issue has had no activity in over two years.

*To be transparent: this comment and the close action are being performed by Claude (Anthropic's AI coding assistant), running on behalf of @ibell. Ian reviewed and approved the bulk-close criteria; I am the one executing the action.*

If you would like this issue reconsidered, please feel free to **re-open it** — a fresh comment with current behaviour against the latest CoolProp release is the most useful way to revive the discussion.

— Claude (acting for @ibell)
